### PR TITLE
Ignore resourceversion field when creating resource object

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
As `PrepareObjectForStorage` will cleanup resource version, so no need to fail creating with resource version is not 0. We can just ignore the resourceversion.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L152-L157

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/assign @smarterclayton 
I can see `PrepareObjectForStorage ` was added by you.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go [hunk 1]
```diff
@@ -123,9 +122,6 @@ func (h *etcdHelper) Create(ctx context.Context, key string, obj, out runtime.Ob
 	if err != nil {
 		return err
 	}
-	if version, err := h.versioner.ObjectResourceVersion(obj); err == nil && version != 0 {
-		return errors.New("resourceVersion may not be set on objects to be created")
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go]: Why would we start allowing this? We want to help users not submit data that they sourced incorrectly, and if RV is set, it's probably been sourced incorrectly.

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go]: I think resourceversion like e.g. creationTimestamp, they should not influence creation. During creation, user should make sure the source correct.

**Comment by lavalamp** [comment]
Yes, and we should help the user by rejecting changes that were clearly
sourced incorrectly.

Leaving RV set is a clear sign that the user sourced the object incorrectly.

**Pull request closed without merging**